### PR TITLE
SWARM-1627: Consolidate MP versions in build-parent.

### PR DESCRIPTION
**Pull request by mkouba**
Motivation
----------
Currently each MP fraction and TCK has versions for the particular spec
within its pom.xml.

Modifications
-------------
Move the versions inside build-parent. Added dependencies into
dependencyManagement section.

Result
------
MP versions are defined in one place.

### CODE REVIEW

File: build-parent/pom.xml [hunk 1]
```diff
@@ -584,6 +595,99 @@
...
+            <scope>test</scope>
+          </dependency>
+
+          <dependency>
+            <groupId>org.eclipse.microprofile.metrics</groupId>
+            <artifactId>microprofile-metrics-api</artifactId>
+            <version>${version.microprofile-metrics}</version>
+          </dependency>
+
+          <dependency>
```
Comment (kenfinnigan) [inline build-parent/pom.xml:636]: Should the TCK dependencies be added to testsuite/microprofile-tck instead?

I presume they're not required for building?

**Comment by kenfinnigan** [comment]
retest this please

**Comment by bobmcwhirter** [comment]
retest this please

**Comment by bobmcwhirter** [comment]
retest this please

**Comment by kenfinnigan** [comment]
test this please

**Comment by heiko-braun** [comment]
@mkouba Looks like this change has side effects: https://ci.wildfly-swarm.io/job/pr-linux-2/3/

File: build-parent/pom.xml (hunk 1 above)
Comment (mkouba) [reply build-parent/pom.xml:636]: No, they're not. I will move them.

**Comment by mkouba** [comment]
@heiko-braun Yep, I forgot to update the `module.xml` files :-(

**Comment by mkouba** [comment]
`AsciidoctorJIntegrationTest` failure: 

> org.openqa.selenium.NoSuchElementException: 
Unable to locate element with name: h2
For documentation on this error, please visit: http://seleniumhq.org/exceptions/no_such_element.html
Build info: version: '2.53.1', revision: 'a36b8b1cd5757287168e54b817830adce9b0158d', time: '2016-06-30 19:26:09'
System info: host: 'slave-linux-04.ci.obsidian.local', ip: '172.16.50.61', os.name: 'Linux', os.arch: 'amd64', os.version: '3.10.0-514.16.1.el7.x86_64', java.version: '1.8.0_151'
Driver info: driver.version: HtmlUnitDriver
	at org.wildfly.swarm.asciidoctorj.test.AsciidoctorJIntegrationTest.testResource(AsciidoctorJIntegrationTest.java:44)

**Comment by heiko-braun** [comment]
I  think we can ignore that for now. @kenfinnigan ?

**Review by kenfinnigan — APPROVED** [review]

**Pull request merged**